discord-bot/bot.js

```diff
@@ -0,0 +1,123 @@
+require('dotenv').config()
+let XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest
+const Discord = require('discord.js')
+const bot = new Discord.Client()
+const token = process.env.TOKEN_OSDC
+const Http = new XMLHttpRequest();
+
+const regex = 'https://imgs.xkcd.com/comics'
+const png = 'png'
+
+bot.on('ready', () => {
+    console.log('The bot is online!!!!')
+})
+
+
+//Greeting message for a New user!!!
+
+bot.on('guildMemberAdd',member => {
+    const channel = member.guild.channels.cache.find(ch => ch.name === 'member-log');
+    if(!channel) return ;
+    message.channel.send(`Hello ${message.author} Welcome to OSDC discord Server..Please introduce yourself!!`)
+})
+
+//help command
+
+bot.on('message',message => {
+        if(message.content === '!help')
+        {
+            const commandEmbedded = new Discord.MessageEmbed()
+            .setTitle('Here are the list of commands ')
+            .addFields(
+                {name : '!help', value: "To view the list of commands " ,inline:false},
+                {name : '!website', value: "Visit our Website" ,inline:false},
+                {name : '!facebook', value: "Follow us on Facebook " ,inline:false},
+                {name : '!twitter', value: "Check us out on Twitter " ,inline:false},
+                {name : '!github', value: "Visit our Github Repository " ,inline:false},
+                {name : '!telegram', value: "Join our telegram channel " ,inline:false},
+                {name : '!xkcd', value: "To get a Comic " ,inline:false},
+                {name : '!irc', value: "Find us on IRC :) " ,inline:false},
+                {name : '!blog', value: "Get the link of OSDC blog" ,inline:false},
+                {name : '!instagram', value: "Follow us on instagram :) " ,inline:false},
+
+            )
+            message.channel.send(commandEmbedded)
+        }
+})
+
+//xkcd comic command
+
```

Comment:
Please write a comment about the logic used to fetch the xkcd comic here. Will help others get familiar with it and contribute to it easily.